src/lib/sql/sql_translator.cpp

```diff
@@ -328,9 +329,8 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_select_statement(cons
       _translate_set_operation(*set_operator);
 
       // In addition to local ORDER BY and LIMIT clauses, the result of the set operation(s) may have final clauses too.
-      if (set_operator->resultOrder) {
-        _translate_order_by(*set_operator->resultOrder);
-      }
+      _translate_order_by_distinct(set_operator->resultOrder, {}, false);
```

Comment:
(Example inspired by the SQL Parser Tests)
```SQL
SELECT *
 FROM store_sales
UNION (SELECT *
         FROM web_sales
        ORDER BY sold_date ASC
        LIMIT 5)
ORDER BY sold_date DESC
LIMIT 10;
```
In this case, `ORDER BY sold_date ASC LIMIT 5` is performed on the second union relation, whereas the whole union has a different order/limit. The first union relation could also have another, different order by/limit.

Reply:
I added an example (with `EXCEPT` instead of `UNION` since we do not support the latter:
https://github.com/hyrise/hyrise/blob/37e0ad7728969900244cccff17b9c1345572edb5/src/lib/sql/sql_translator.cpp#L330-L341